Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+        String FILE_NAME = "mytest.txt";
+
+        // Create job run at the specified pool
+        PoolInformation poolInfo = new PoolInformation();
+        poolInfo.setPoolId(poolId);
+        client.getJobOperations().createJob(jobId, poolInfo);
+
+        // Create task
+        TaskAddParameter taskToAdd = new TaskAddParameter();
+        taskToAdd.setId("mytask");
+        taskToAdd.setCommandLine(String.format("cat %s", FILE_NAME));
+
+        String sas = uploadFileToCloud(container);
+
+        // Associate resource file with task
+        ResourceFile file = new ResourceFile();
+        file.setFilePath(FILE_NAME);
+        file.setBlobSource(sas);
+        List<ResourceFile> files = new ArrayList<ResourceFile>();
+        files.add(file);
+        taskToAdd.setResourceFiles(files);
+
+        // Add task to job
+        client.getTaskOperations().createTask(jobId, taskToAdd);
+    }
+
+    // Wait all tasks under a specified job to be completed
+    private static boolean waitForTasksToComplete(BatchClient client, String jobId, Duration expiryTime) throws BatchErrorException, IOException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+
+        while (elapsedTime < expiryTime.toMillis()) {
+            List<CloudTask> taskCollection = client.getTaskOperations().listTasks(jobId, new DetailLevel.Builder().selectClause("id, state").build());
+
+            boolean allComplete = true;
+            for (CloudTask task : taskCollection) {
+                if (task.getState() != TaskState.COMPLETED) {
+                    allComplete = false;
+                    break;
+                }
+            }
+
+            if (allComplete) {
+                // All tasks completed
+                return true;
+            }
+
+            System.out.println("wait 10 seconds for tasks to complete...");
+
+            // Check again after 10 seconds
+            Thread.sleep(10 * 1000);
+            elapsedTime = (new Date()).getTime() - startTime;
+        }
+
+        // Timeout, return false
+        return false;
+    }
+
+    public static void main(String argv[]) throws Exception {
+
+        // Get batch and storage account information from environment
+        String batchAccount = System.getenv("AZURE_BATCH_ACCOUNT");
+        String batchKey = System.getenv("AZURE_BATCH_ACCESS_KEY");
+        String batchUri = System.getenv("AZURE_BATCH_ENDPOINT");
+
+        String storageAccountName = System.getenv("STORAGE_ACCOUNT_NAME");
+        String storageAccountKey = System.getenv("STORAGE_ACCOUNT_KEY");
+
+        Boolean shouldDeleteContainer = true;
+        Boolean shouldDeleteJob = true;
+        Boolean shouldDeletePool = false;
+
+        Duration TASK_COMPLETE_TIMEOUT = Duration.ofMinutes(10);
```

Comment:
This seems really long given that we waited for 1 VM in the pool to be up before even waiting on the tasks.  I would expect something more like 1m max.